lib/webmushra/pages/LikertSingleStimulusPage.js

```diff
@@ -106,31 +121,33 @@ LikertSingleStimulusPage.prototype.save = function () {
   this.fpc.unbind(); 
   this.time += (new Date() - this.startTimeOnPage);
   
-  this.result = $("input[name='"+this.likert.prefix +"_response']:checked").val();
+  for (var i = 0; i < this.likerts.length; ++i) {
+    this.results[i] = $("input[name='"+this.likerts[i].prefix +"_response']:checked").val();
+  }
   
   this.filePlayer.free();
 };
 
 LikertSingleStimulusPage.prototype.store = function (_reponsesStorage) {
-		
-	
-	
   var trial = this.session.getTrial(this.pageConfig.type, this.pageConfig.id);
   if (trial === null) {
     trial = new Trial();
-	trial.type = this.pageConfig.type;
-	trial.id = this.pageConfig.id;
-	this.session.trials[this.session.trials.length] = trial;	  
+    trial.type = this.pageConfig.type;
+    trial.id = this.pageConfig.id;
+    this.session.trials[this.session.trials.length] = trial;
   }
   var rating = new LikertSingleStimulusRating();
   rating.stimulus = this.stimulus.id;
   
-  if(this.result == undefined){
-    rating.stimulusRating = "NA";
-  }else{
-    rating.stimulusRating = this.result;
+  rating.stimulusRating = [];
+  for(var i = 0; i < this.likerts.length; ++i){
+    if(this.results[i] == undefined){
```

Comment:
`this.results[i] === undefined`

Reply:
...while you are at it ;-)

Reply:
Obviously better and happy to change it, but I just want to flag that I just copied that from existing code. Speaking of which, the way the grunt jshint is set up flags usage of modern JS constructs like =>, which I subsequently removed from my code in favour of older syntax, but it would be nice if the toolchain were updated to something more modern.

Reply:
👍 in hindsight we would have been better off with a modern js framework. The decision to write this from scratch was maybe not the best. But my guess it that rewrite all of this for vue or react would be way too much work :-(